catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/XMLSaxEventHandlerImpl.java

```diff
@@ -0,0 +1,186 @@
+     * @param attributes the attributes that are passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     */
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) {
+        /*
+         * If the element name is in the xmlToMetacardMapping, begin elementBeingRead. The data will be in the characters event.
+         */
+        if (xmlToMetacard.get(localName.toLowerCase()) != null) {
+            elementBeingRead = localName.toLowerCase();
+            return;
+        }
+
+        /*
+         * If the attribute "name" is in the xmlToMetacardMapping, take the "name" value, and add it directly to to the attributes list
+         */
+        String attribute = attributes.getValue("name");
+        if (attribute != null && xmlToMetacard.get(attribute) != null) {
+            elementBeingRead = attribute;
+        }
+    }
+
+    /**
+     * Takes in a sax event from {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}. If the element is in the xmlToElementMapping,
+     * add it to the attributes list
+     *
+     * @param namespaceURI the namespaceURI that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param localName    the localName that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param qName        the qName that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     */
+    @Override
+    public void endElement(String namespaceURI, String localName, String qName) {
+        /*
+         * If the handler is currently "reading" an element and the element is in the xmlToElementMapping, add the value of the
+         * textDataOfElement to the attributes list
+         */
+        if (elementBeingRead != null) {
+            String result = textDataOfElement.toString().trim();
+            textDataOfElement.setLength(0);
+            if (xmlToMetacard.get(elementBeingRead) != null) {
+                attributes.add(new AttributeImpl(xmlToMetacard.get(elementBeingRead), result));
+            }
+            elementBeingRead = null;
+
+        }
+    }
+
+    /**
+     * Takes in a sax event from {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate} and adds the characters
+     * to textDataOfElement
+     *
+     * @param ch     the ch that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param start  the start that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param length the length that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     */
+    @Override
+    public void characters(char[] ch, int start, int length) {
+        if (elementBeingRead != null) {
```

Comment:
What does it mean if the elementBeingRead is null? Does it mean we are not currently reading an element, we haven't started reading anything, or something else? 

I only see two places where we check `elementBeingRead != null`, but it might be worth it to replace that with a predicate method like:

```
boolean isElementBeingRead() {
    return elementBeingRead != null;
}
```

or whatever name is appropriate like `startredReading`, `stillReading`, `nowReading` ...